### PR TITLE
Add the terminology and try and clarify everywhere to specify pre-loa…

### PR DESCRIPTION
**Pull request by tomrittervg**
…ded when necessary.

### CODE REVIEW

File: draft-ietf-trans-gossip.md [hunk 1]
```diff
@@ -401,30 +409,43 @@ with the following content:
...
     {{RFC-6962-BIS-09}} Section 3.3.
 
-The 'x509\_chain' element MUST contain at least the leaf certificate
-and SHOULD contain the full chain to a root accepted by all of the
-logs in the set of logs issuing all the SCTs in the 'sct\_data'
-element.
+The 'x509\_chain' element MUST contain the full chain from the leaf certificate 
+to a trust anchor (either pre-loaded or locally added) accepted by all of the 
+logs in the set of logs issuing all the SCTs in the 'sct\_data' element.
+
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:415]: This change makes clients forced to expose more than necessary for CT gossip to fulfill its task of catching lying logs.
A client that wishes to help exposing a lying CA should make arrangements for the full chain to be included in a log.

File: draft-ietf-trans-gossip.md [hunk 2]
```diff
@@ -401,30 +409,43 @@ with the following content:
...
+  "a trust anchor accepted by all of the logs in the set of logs issuing all 
+    the SCTs in the 'sct\_data' element."
+  The client doesn't know if the log accepts the root it chained to. All it knows is
+  "This is an SCT from a log I trust." and "This is the cert chain I built."
+
+  What was the rational for keeping that phrase in? Shouldn't it just be: 
+  The 'x509\_chain' element MUST contain the full chain from the leaf certificate 
+  to a trust anchor.
+\]
+
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:427]: The rationale is that this protects exposure of locally added logs and/or CA:s. Can't remember if both but I'm sure dkg knows! Is he here? Should I move this discussion back to the draft text?

Also, the client does know that a log which has issued an SCT for a cert accepts the root it's chained to -- if not, the log wouldn't have accepted it and thus not issued an SCT.

File: draft-ietf-trans-gossip.md [hunk 3]
```diff
@@ -401,30 +409,43 @@ with the following content:
...
+Feedback.
 
-Some clients have trust anchors that are locally added (e.g. by an
-administrator or by the user themselves). A local trust anchor is
-potentially privacy-sensitive since it may carry information about the
-specific computer or user. If a certificate is covered by SCTs issued
-by publicly trusted logs, but it chains to a privacy-sensitive local
-trust anchor, the client SHOULD NOT include anything but the leaf
-certificate in the 'x509\_chain' element.
+If a certificate is validated by SCTs issued by publicly trusted logs, but
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:438]: "publicly trusted logs" is not well defined. Do you mean "pre-loaded logs"?

File: draft-ietf-trans-gossip.md [hunk 4]
```diff
@@ -401,30 +409,43 @@ with the following content:
...
-Some clients have trust anchors that are locally added (e.g. by an
-administrator or by the user themselves). A local trust anchor is
-potentially privacy-sensitive since it may carry information about the
-specific computer or user. If a certificate is covered by SCTs issued
-by publicly trusted logs, but it chains to a privacy-sensitive local
-trust anchor, the client SHOULD NOT include anything but the leaf
-certificate in the 'x509\_chain' element.
+If a certificate is validated by SCTs issued by publicly trusted logs, but 
+chains to a local trust anchor, the client MAY perfom SCT Feedback 
+for this SCT and certificate chain bundle. Perfoming SCT Feedback may be
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:440]: s/chain bundle/chain/1 should be fine, right?

File: draft-ietf-trans-gossip.md [hunk 5]
```diff
@@ -515,12 +536,18 @@ An HTTPS client will retrieve SCTs from an HTTPS server, and must
...
 historical STH is considered personally identifiable information per
 above, the client must obtain a consistency proof to a more recent
 STH.
 
+A client SHOULD perform proof fetching. A client MUST NOT perform 
+proof fetching for any SCTs or STHs issued by a locally added log. 
+A client MAY fetch an inclusion proof for an SCT (issued by a 
+pre-loaded log) that validates a certificate chaining to a locally 
+added trust anchor.
+
```
Comment (ln5) [inline draft-ietf-trans-gossip.md:550]: Isn't the MUST NOT too strong? I run my own log, or my favourite digital rights organisation do, and I configure my browser to use it. Should this fall in the Trusted Auditor Relationship?

File: draft-ietf-trans-gossip.md (hunk 1 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:415]: I think this is a change from what we talked about last call... We can talk about it more this call.

File: draft-ietf-trans-gossip.md (hunk 3 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:438]: Yes.

File: draft-ietf-trans-gossip.md (hunk 4 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:440]: Yes.

File: draft-ietf-trans-gossip.md (hunk 5 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:550]: What would proof fetching give you?  You can resolve a SCT to a STH - but you can't pollinate the STH.  You can resolve a STH to a new STH - but again, you can't pollinate it.

File: draft-ietf-trans-gossip.md (hunk 2 above)
Comment (tomrittervg) [inline draft-ietf-trans-gossip.md:427]: > The rationale is that this protects exposure of locally added logs and/or CA:s. Can't remember if both but I'm sure dkg knows! Is he here? Should I move this discussion back to the draft text?

I'm not sure I grok this fully. We can talk about it on the call.

> Also, the client does know that a log which has issued an SCT for a cert accepts the root it's chained to -- if not, the log wouldn't have accepted it and thus not issued an SCT.

Not necessarily. a) The log may have accepted the cert when it chained up to a _different_ root than the one the client built a path to.  b) The log may have trusted the root at the point in time the cert was entered, but not longer does.  Again, call.

File: draft-ietf-trans-gossip.md (hunk 5 above)
Comment (ln5) [inline draft-ietf-trans-gossip.md:550]: It'd give me the proof I wanted. :) Maybe the issue here is that a gossip draft forbids a client to do something useful outside the scope of gossip.

File: draft-ietf-trans-gossip.md (hunk 4 above)
Comment (ln5) [inline draft-ietf-trans-gossip.md:440]: fixed

File: draft-ietf-trans-gossip.md (hunk 3 above)
Comment (ln5) [inline draft-ietf-trans-gossip.md:438]: fixed

**Comment by tomrittervg** [comment]
This will be redone following our discussion today.

**Comment by ln5** [comment]
You could rebase on https://github.com/ln5/draft-ietf-trans-gossip/tree/tjr-preloaded-locally-added to pick up my small changes if you wanted to.

**Pull request closed without merging**